src/ai/backend/manager/registry.py

```diff
@@ -2982,6 +2985,61 @@ async def mark_session_terminated(
     ) -> None:
         await self.clean_session(session_id)
 
+    async def get_commit_status(
+        self,
+        session_name_or_id: Union[str, SessionId],
+        access_key: AccessKey,
+    ) -> bool:
+        kernel = await self.get_session(session_name_or_id, access_key)
+
+        async with self.handle_kernel_exception("commit_session", kernel["id"], access_key):
+            async with RPCContext(
+                kernel["agent"],
+                kernel["agent_addr"],
+                invoke_timeout=None,
+                order_key=kernel["id"],
+                keepalive_timeout=self.rpc_keepalive_timeout,
+            ) as rpc:
+                return await rpc.call.get_commit_status(str(kernel["id"]))
+
+    async def commit_session(
+        self,
+        session_name_or_id: Union[str, SessionId],
+        access_key: AccessKey,
+        filename: str | None,
+    ) -> Mapping[str, Any]:
+        """
+        Commit a main kernel's container of the given session.
+        """
+
+        kernel = await self.get_session(session_name_or_id, access_key)
+        if filename is None:
+            now = datetime.now(tzutc())
+            filename = f"{now.isoformat()}.tar.gz"
+        assert filename is not None
+
+        async with self.db.begin_readonly() as db_conn:
+            query = (
+                sa.select([users.c.email])
+                .select_from(users)
+                .where(users.c.uuid == kernel["user_uuid"])
+            )
+            result = await db_conn.execute(query)
+            user_email = result.scalar()
+        # TODO: get path from toml or cfg
+        path = Path(str(user_email), str(kernel["session_id"]), filename)
```

Comment:
Is it safe to directly use email and ISO-formatted datetime string as path components?

Reply:
How is this value related with the agent-side commit path configuration as you noted as comments?

Reply:
Id like to make sure admin can check whose commits they are.
agent-side commit path is the root of commit path and this email and ISO formatted datetime path is underlying path.

the path would be like the.
./tmp/backend.ai/commit/myemail@email/20220801